From d8b779513474ece1f5ec2fc9b7c32afb73e32181 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Mon, 29 Apr 2019 10:14:12 -0400
Subject: [PATCH] daemon: ensure cache files for system users are processed

At the moment we skip cache files for system users.  That
doesn't make much sense; if there's a cache file we should
be using it.

This commit changes the code to read cache files, even for
system users, and so lets root have a non-default session.

Closes: https://gitlab.freedesktop.org/accountsservice/accountsservice/issues/65
---
 src/daemon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: accountsservice-0.6.55/src/daemon.c
===================================================================
--- accountsservice-0.6.55.orig/src/daemon.c	2019-05-30 00:12:49.993056439 +0200
+++ accountsservice-0.6.55/src/daemon.c	2019-05-30 00:12:50.289059484 +0200
@@ -486,7 +486,7 @@ reload_users (Daemon *daemon)
         load_entries (daemon, users, TRUE, entry_generator_requested_users);
 
         /* Now add/update users from other sources, possibly non-local */
-        load_entries (daemon, users, FALSE, entry_generator_cachedir);
+        load_entries (daemon, users, TRUE, entry_generator_cachedir);
 
         wtmp_helper_update_login_frequencies (users);
 
